### PR TITLE
Allow commands to abort pipeline execution

### PR DESCRIPTION
**Pull request by kevsmith**
Implements Relay parts of the design described in [#691](https://github.com/operable/cog/issues/691).

Fixes [#691](https://github.com/operable/cog/issues/691)

### CODE REVIEW

File: relay/worker/output_parser.go [hunk 1]
```diff
@@ -58,7 +58,11 @@ func parseOutput(result api.ExecResult, err error, resp *messages.ExecutionRespo
 		return
 	}
 
-	resp.Status = "ok"
+	if resp.Aborted == true {
+		resp.Status = "abort"
+	} else {
+		resp.Status = "ok"
```
Comment (christophermaier) [inline relay/worker/output_parser.go]: Does this mean that the status is still "OK" even if an error occurred?

File: relay/worker/output_parser.go [hunk 2]
```diff
@@ -111,3 +115,15 @@ func extractTemplate(line []string, resp *messages.ExecutionResponse, req messag
 func flagJSON(line []string, resp *messages.ExecutionResponse, req messages.ExecutionRequest) {
 	resp.IsJSON = true
 }
+
+func parseAction(line []string, resp *messages.ExecutionResponse, req messages.ExecutionRequest) {
+	if len(line) < 2 {
```
Comment (christophermaier) [inline relay/worker/output_parser.go]: What's the magic number 2?

Comment (kevsmith) [inline relay/worker/output_parser.go]:
``` go
package main

import (
    "fmt"
    "regexp"
)

func main() {
    r := regexp.MustCompilePOSIX("^COGCMD_ACTION:")
    text := "COGCMD_ACTION: abort"
    fmt.Printf("Parsing: %s\n", text)
    for i, v := range r.Split(text, 2) {
        fmt.Printf("text[%d] = %s\n", i, v)
    }
    text = "COGCMD_ACTION:"
    fmt.Printf("Parsing: %s\n", text)
    for i, v := range r.Split(text, 2) {
        fmt.Printf("text[%d] = %s\n", i, v)
    }

}
```

I thought the above code (analogous to how Relay currently parses command output) would produce the following output:

```
Parsing: COGCMD_ACTION: abort
text[0] = 
text[1] =  abort
Parsing: COGCMD_ACTION:
text[0] =
```

Meaning that `Split` would return an array w/2 or 1 member depending on if the string contained content past the matching regex. Turns out I was wrong at least wrt to Go's behavior. In reality the above code produces:

```
Parsing: COGCMD_ACTION: abort
text[0] =
text[1] =  abort
Parsing: COGCMD_ACTION:
text[0] =
text[1] =
```

tl;dr You found a bug :)

File: relay/worker/output_parser.go (hunk 1 above)
Comment (kevsmith) [inline relay/worker/output_parser.go]: I don't think that could happen given that the clauses before this one return early in the case of an error. I can clean up this logic to make it more evident.
